app/server/update-controller/src/main/java/io/syndesis/server/update/controller/impl/ConnectorUpdateHandler.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright (C) 2016 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.syndesis.server.update.controller.impl;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+import io.syndesis.common.model.ChangeEvent;
+import io.syndesis.common.model.Kind;
+import io.syndesis.common.model.bulletin.ConnectionBulletinBoard;
+import io.syndesis.common.model.bulletin.LeveledMessage;
+import io.syndesis.common.model.connection.Connection;
+import io.syndesis.common.model.connection.Connector;
+import io.syndesis.server.dao.manager.DataManager;
+import io.syndesis.server.update.controller.AbstractResourceUpdateHandler;
+import io.syndesis.server.update.controller.ResourceUpdateHelper;
+
+
+public class ConnectorUpdateHandler extends AbstractResourceUpdateHandler<ConnectionBulletinBoard> {
+    private final List<Kind> supportedKinds;
+
+    public ConnectorUpdateHandler(DataManager dataManager) {
+        super(dataManager);
+
+        this.supportedKinds = Arrays.asList(Kind.Connector, Kind.Connection);
+    }
+
+    @Override
+    public boolean canHandle(ChangeEvent event) {
+        return event.getKind().map(Kind::from).filter(supportedKinds::contains).isPresent();
+    }
+
+    @Override
+    protected List<ConnectionBulletinBoard> compute(ChangeEvent event) {
+        final List<ConnectionBulletinBoard> boards = new ArrayList<>();
+        final DataManager dataManager = getDataManager();
+        final List<Connection> connections = dataManager.fetchAll(Connection.class).getItems();
+        final List<Connector> connectors = dataManager.fetchAll(Connector.class).getItems();
+
+        for (int i = 1; i < connectors.size(); i++) {
+            final Connector connector = connectors.get(i);
+            final String id = connector.getId().get();
+
+            for (int c = 0; c < connections.size(); c++) {
```

Comment:
O(N^2) = Could get ugly one day.

Reply:
This part is everything but optimized I'm working to use DataManager::fetchIdsByPropertyValue